pkg/controller/infrastructure/actuator.go

```diff
@@ -31,6 +32,13 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
+// InfrastructureState represents the last known State of an Infrastructure resource.
+// It is saved after a reconciliation and used during restore operations.
+type InfrastructureState struct {
+	SavedProviderStatus *runtime.RawExtension `json:"savedProviderStatus,omitempty"`
```

Comment:
doc strings for exported properties. Sorry for missing that in my first review.

Reply:
Sorry completely forgot about that as well. Add the comments now.